cli/main.py

```diff
@@ -0,0 +1,172 @@
+
+
+@click.command()
+@click.argument("secret_file", type=click.Path(exists=True))
+@click.option(
+    "-d", "--data", "view_data", type=click.STRING, multiple=True, default=None
+)
+def view_secret(secret_file, view_data):
+    """Command to view sealed secrets.
+
+    :option: data - Option for viewing a secret actual value by providing secret name
+
+    Example: python cli/main.py view-secret cap/environments/cap-qa/sealedsecrets/cap-creds.yml
+    Example: python cli/main.py view-secret cap/environments/cap-qa/sealedsecrets/cap-creds.yml -d FLOWER_USER -d FLOWER_PASSWORD
+    """
+    logging.info(
+        "Reading yml file...",
+    )
+    with open(secret_file, "r") as tmp:
+        data = yaml.safe_load(tmp)
+    if not data:
+        logging.error("No data present. Please check your secret file!")
+        sys.exit(1)
+
+    namespace = data.get("metadata").get("namespace")
+    secret_name = data.get("metadata").get("name")
+    base64_data = get_base64data(namespace, secret_name)
+    base64_data_keys = base64_data.get("data").keys()
+
+    if view_data:
+        base64_data_keys = [
+            key for key in view_data if validate_key(base64_data_keys, key)
+        ]
+
+    for key in base64_data_keys:
+        secret_value = base64.b64decode(base64_data.get("data").get(key)).decode(
+            "utf-8"
+        )
+        click.echo(f"{key}:{secret_value}")
+
+
+@click.command()
+@click.argument("secret_file", type=click.Path(exists=True))
+@click.option(
+    "-d", "--data", "update_data", type=click.STRING, multiple=True, default=None
+)
+def update_secret(secret_file, update_data):
+    """Command to update sealed secrets.
+
+    Update the Sealed secret with new value.
+    :option: data - Option for updating a secret actual value by providing secret name in key value pair
+
+    Example: python cli/main.py update-secret cap/environments/cap-qa/sealedsecrets/cap-creds.yml -d FLOWER_USER:NEW_VAL -d FLOWER_PASSWORD:MY_VAL
+    """
+    logging.info(
+        "Reading yml file...",
+    )
+    with open(secret_file, "r") as tmp:
+        data = yaml.safe_load(tmp)
+    if not data:
+        logging.error("No data present. Please check your secret file!")
+        sys.exit(1)
+
+    namespace = data.get("metadata").get("namespace")
+    secret_name = data.get("metadata").get("name")
+    base64_data = get_base64data(namespace, secret_name)
+    base64_data_keys = base64_data.get("data").keys()
+
+    if not update_data:
+        logging.error("Please provide keys to update!")
+        sys.exit(1)
+
+    requested_change = dict(secret.split(":", 1) for secret in update_data)
+    for k, v in requested_change.items():
+        if validate_key(base64_data_keys, k):
+            update_key(base64_data, k, v)
+
+    # Keeping required metadata only
+    base64_data["metadata"] = {"name": secret_name, "namespace": namespace}
+    base64_data.pop("type", None)
+
+    with open("tmp", "w") as tmp:
+        yaml.dump(base64_data, tmp)
+
```

Comment:
We shouldn't run `kubeseal` if there is a `validate_key` error. Only when there are 0 errors.